src/resources/assets/js/mixins/largoContainer.vue

```diff
@@ -81,27 +81,26 @@ export default {
             return this.step === 1;
         },
         annotations() {
-            if (!this.selectedLabel){
+            if (!this.selectedLabel) {
                 return [];
             }
 
-            let filterLabel = { label: this.selectedLabel.id };
+            let cacheKey = JSON.stringify({ label: this.selectedLabel.id });
 
-            if (this.selectedFilters.length > 0){
-                filterLabel["filters"] = this.selectedFilters;
-                filterLabel["union"] = this.union;
-            }
+            if (this.annotationsCache.hasOwnProperty(cacheKey)) {
+                let annotations = this.annotationsCache[cacheKey];
+                let filtersCacheKey = JSON.stringify({...this.selectedFilters, label: this.selectedLabel.id, union: this.union})
 
-            let cacheKey = JSON.stringify(filterLabel);
+                if (this.hasActiveFilters && this.filtersCache.hasOwnProperty(filtersCacheKey)) {
+                    annotations = annotations.filter(annotation => this.filtersCache[filtersCacheKey].includes(annotation.id));
```

Comment:
This looks a little inefficient, since `includes()` could traverse the whole array each time. What about adding an object/map instead of an array to `filtersCache` when it is populated (`{id: true}`)? Then you could do O(1) lookups here (`this.filtersCache[filtersCacheKey][annotation.id] === true`).

I'm a little sensitive about performance here because you can have hundreds of thousands of annotations in Largo.